src/components/Tamagotchi.tsx

```diff
@@ -1,28 +1,70 @@
 "use client";
 import { INTERACTION } from "@/app/utils/interaction";
 import React, { useState, useEffect } from "react";
-import { idle, drinkingCoffee } from "./tamagotchiFrames";
+import { idle, drinkingCoffee, death } from "./tamagotchiFrames";
 
 import "../index.css";
 
 const DEFAULT_STATUS = ":)";
 const Tamagotchi: React.FC = () => {
   const [frameIndex, setFrameIndex] = useState<number>(0);
-  const [animation, setAnimation] = useState(idle);
+  const [tamagotchiState, setTamagotchiState] = useState<any>({});
+  const [animation, setAnimation] = useState<string[]>(idle);
   const [tamaStatus, setTamaStatus] = useState(DEFAULT_STATUS);
+  const [checkingStatus, setCheckingStatus] = useState(false);
+  const [isInteracting, setIsInteracting] = useState(false);
 
   //TODO - call init endpoint to determine if tamagotchi is initialized. if not generate one.
 
   useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await fetch("/api/getState", { method: "POST" });
+        if (response.ok) {
+          const jsonData = await response.json();
+          setTamagotchiState(jsonData);
+          handleDeath(jsonData, pollInterval);
+        }
+      } catch (error) {
+        console.error("Error fetching data:", error);
+      }
+    };
+
     // Cycle through frames every 1 second
-    const interval = setInterval(() => {
+    const frameInterval = setInterval(() => {
       setFrameIndex((prevIndex) => (prevIndex + 1) % idle.length);
     }, 1000);
 
-    return () => clearInterval(interval);
+    // Fetch data on component mount
+    fetchData();
+
+    // Start polling data every N seconds (adjust the interval as needed)
+    const pollInterval = setInterval(fetchData, 5000); // Poll every 5 seconds
+
+    return () => {
+      clearInterval(pollInterval);
+      clearInterval(frameInterval);
+    };
   }, []);
 
+  const handleResponse = (responseText: string) => {
+    const responseJSON = JSON.parse(responseText);
+    const animation = JSON.parse(responseJSON.animation);
+    const status = responseJSON.status;
+    setFrameIndex(0);
+    setAnimation(animation);
+    setTamaStatus(status);
+  };
+
+  const handleDeath = (jsonData: any, pollInterval: NodeJS.Timer) => {
+    if (jsonData.death) {
+      setAnimation(death);
+      setTamaStatus("Dead :(");
+      clearInterval(pollInterval);
+    }
+  };
   const feedTamagotchi = async (e: any) => {
```

Comment:
**issue (complexity):** Consider consolidating the repetitive interaction handlers into a single generic handler function.

The interaction handlers follow identical patterns and can be simplified into a single generic handler while preserving all functionality. Consider:

```typescript
const handleInteraction = async (interactionType: INTERACTION, statusText: string) => {
  setIsInteracting(true);
  setTamaStatus(statusText);

  try {
    const response = await fetch("/api/interact", {
      method: "POST",
      body: JSON.stringify({ interactionType }),
      headers: { "Content-Type": "application/json" },
    });
    handleResponse(await response.text());
  } catch (e) {
    console.log(e);
  }

  setTimeout(() => {
    setAnimation(idle);
    setTamaStatus(DEFAULT_STATUS);
    setIsInteracting(false);
  }, 9000);
};
```

Then simplify all handlers to:

```typescript
const feedTamagotchi = () => handleInteraction(INTERACTION.FEED, "Feeding...");
const playWithTamagotchi = () => handleInteraction(INTERACTION.PLAY, "Playing...");
const treatSickTamagotchi = () => handleInteraction(INTERACTION.GO_TO_HOSPITAL, "Treating...");
const handleDiscipline = () => handleInteraction(INTERACTION.DISCIPLINE, "Disciplining...");
```

This maintains all functionality while significantly reducing code duplication and complexity.